### PR TITLE
Fixes #9255 - Explain now to handle empty strings and nulls

### LINKED ISSUE #9255: Switch test for $null and empty string
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Typo in `Get-Foo` cmdlet" instead of "Typo."
- [X] **Verify Version:** If there is a mismatch between documentation and the behavior on your system, ensure that the version you are using is the same as the documentation. Check this box if they match or the issue you are reporting is not version specific.

### Links

- https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-switch

I have not confirmed other versions, but assume this has always been the case.

### Summary

The documentation states:

> Notice that the empty string value doesn't match $null but $null matches both $null and an empty string.

Testing reveals that the case for an empty string also matches $null.  A switch statement cannot distinguish between an empty string and $null. 

### Details

Modify the given example like this:

```
$values = '', 5, $null
switch ( $values )
{
    ''      { "Value '$_' is an empty string" ; continue }
    $null   { "Value '$_' is `$null or an empty string" }
    default { "Value [$_] is not an empty string or `$null" }
}

```

The output shows that '' also matches $null and prevents testing for $null:

> Value '' is an empty string
> Value [5] is not an empty string or $null
> Value '' is an empty string
> 

### Suggested Fix

I you must distinguish between empty string and null, use this method:

```
$values = '', 5, $null
switch ( $values )
{
	{ $null -eq $_ }  { "Value '$_' is `$null" ; continue }
    ''      { "Value '$_' is an empty string" }
    default { "Value [$_] is not an empty string or `$null" }
}

```
The successful test output is:

> Value '' is an empty string
> Value [5] is not an empty string or $null
> Value '' is $null

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes AB#9110 - Fixes #9255 - Explain now to handle empty strings and nulls

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/docs-conceptual/learn/deep-dives/everything-about-switch.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ $result
 ```
 
 For this example, the value of `$day` matches one of the numeric values, then the correct name is
-assigned to `$result`. We are only doing a variable assignment in this example, but any PowerShell
+assigned to `$result`. we're only doing a variable assignment in this example, but any PowerShell
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/everything-about-switch.md:82]:
```suggestion
assigned to `$result`. We're only doing a variable assignment in this example, but any PowerShell
```

File: reference/docs-conceptual/learn/deep-dives/everything-about-switch.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ $result = switch ( $day )
 }
 ```
 
-We are placing the value on the PowerShell pipeline and assigning it to the `$result`. You can do
+we're placing the value on the PowerShell pipeline and assigning it to the `$result`. You can do
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/everything-about-switch.md:102]:
```suggestion
We're placing the value on the PowerShell pipeline and assigning it to the `$result`. You can do
```

File: reference/docs-conceptual/learn/deep-dives/everything-about-switch.md [hunk 3]
```diff
@@ -560,7 +560,7 @@ It still executes the same way and gives a better visual break when quickly look
 
 ### Regex $matches
 
-We need to revisit regex to touch on something that isn't immediately obvious. The use of regex
+We need to revisit regex to touch on something that'sn't immediately obvious. The use of regex
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/everything-about-switch.md:563]:
```suggestion
We need to revisit regex to touch on something that isn't immediately obvious. The use of regex
```

File: reference/docs-conceptual/learn/deep-dives/everything-about-switch.md [hunk 4]
```diff
@@ -630,7 +647,7 @@ switch ( $file )
 ### Constant expression
 
 Lee Dailey pointed out that we can use a constant `$true` expression to evaluate `[bool]` items.
-Imagine if we have several boolean checks that need to happen.
+Imagine if we've several boolean checks that need to happen.
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/everything-about-switch.md:650]:
```suggestion
Imagine if we have several boolean checks that need to happen.
```

File: reference/docs-conceptual/learn/deep-dives/everything-about-switch.md [hunk 5]
```diff
@@ -724,7 +741,7 @@ access that value directly. I would call it madness.
 ### Hashtables
 
 One of my most popular posts is the one I did on [hashtables][hashtables]. One of the use cases for
-a `hashtable` is to be a lookup table. That is an alternate approach to a common pattern that a
+a `hashtable` is to be a lookup table. that's an alternate approach to a common pattern that a
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/everything-about-switch.md:744]:
```suggestion
a `hashtable` is to be a lookup table. That's an alternate approach to a common pattern that a
```
